### PR TITLE
IGNITE-19425 Use data nodes from DistributionZoneManager instead of BaselineManager#nodes and remove BaselineManager

### PR DESCRIPTION
**Pull request by alievmirza**
https://issues.apache.org/jira/browse/IGNITE-19425

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 1]
```diff
@@ -288,6 +289,24 @@ public CompletableFuture<Set<String>> dataNodes(long causalityToken, int zoneId)
...
     }
 
+    /**
+     * Returns the actual data nodes of the specified zone.
+     *
+     * @param zoneId Zone id.
+     * @return The future which will be completed with data nodes for the zoneId.
+     */
+    public CompletableFuture<Set<String>> currentDataNodes(int zoneId) {
+        return metaStorageManager.get(zoneDataNodesKey(zoneId)).thenApply(dataNodesEntry -> {
```
Comment (sanpwc) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:299]: So, you're mixing distributively current dataNodes with locally current zone cfg, right? That seems incorrect.

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 2]
```diff
@@ -288,6 +289,24 @@ public CompletableFuture<Set<String>> dataNodes(long causalityToken, int zoneId)
...
+            int catalogVersion = catalogManager.latestCatalogVersion();
```
Comment (sanpwc) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:300]: It's incorrect. Catalog latest version might be one step below over local dataNodes. Please use causality token bounded dataNodes retrieval in order to match catalog and dataNodes.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java [hunk 3]
```diff
@@ -306,6 +307,8 @@ private void doOnNewPeersConfigurationApplied(PeersAndLearners configuration) {
                     )
             ).get();
 
+            Set<Assignment> calculatedAssignments = calculateAssignmentsFn.apply(tablePartitionId).get();
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java:310]: So, just blocking get(). Nice way to get a deadlock.

Comment (sanpwc) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java:310]: > We have already make get on ms.getAll in a few lines above
Yep, and we also have `// TODO: https://issues.apache.org/jira/browse/IGNITE-17592` Remove synchronous wait in a few more lines above.
Within which thread are we going to wait? Is it a raft one?

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java (hunk 2 above)
Comment (alievmirza) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:300]: I've removed this method

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java (hunk 1 above)
Comment (alievmirza) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:299]: I've removed this method

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java (hunk 3 above)
Comment (alievmirza) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java:310]: No, this is a thread from `RebalanceRaftGroupEventsListener#rebalanceScheduler`

**Review by sashapolo — COMMENTED** [review]
I'm ok with test classes renaming, thank you

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 4]
```diff
@@ -1455,17 +1448,19 @@ private void dropTableLocally(long causalityToken, CatalogTableDescriptor tableD
...
-                tablePartitionId.partitionId(),
-                getZoneDescriptor(tableDescriptor, catalogVersion).replicas()
-        );
+        CatalogZoneDescriptor zoneDescriptor = getZoneDescriptor(tableDescriptor, catalogVersion);
+
+        return distributionZoneManager.dataNodes(zoneDescriptor.updateToken(), tableDescriptor.zoneId()).thenApply(dataNodes ->
+                AffinityUtils.calculateAssignmentForPartition(
+                        dataNodes,
+                        tablePartitionId.partitionId(),
+                        getZoneDescriptor(tableDescriptor, catalogVersion).replicas()
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1462]: Seems that we already have zoneDesciptor variable, right?

Comment (alievmirza) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java]: Fixed

**Review by sanpwc — APPROVED** [review]

**Pull request merged**